nsm/scripts/generate-routes.js

```diff
@@ -5,7 +5,144 @@ const prettier = require("prettier")
 const { existsSync } = require("fs")
 const fs = require("fs/promises")
 
-async function generateRoutes() {
+async function generateNsmPagesManifest(rootDir, outputDir) {
+  const pagesDir = existsSync(path.resolve(rootDir, "pages"))
+    ? path.resolve(rootDir, "pages")
+    : path.resolve(rootDir, "src/pages")
+  const pageFiles = glob.sync("**/*.{ts,tsx,js,jsx}", { cwd: pagesDir })
+
+  const manifest = {}
+
+  for (const pageFile of pageFiles) {
+    const relativePagePath = path.relative(process.cwd(), pageFile)
+    const parsedPagePath = path.parse(relativePagePath)
+
+    const isInvalidExtension = ![".tsx", ".jsx", ".ts", ".js"].includes(
+      parsedPagePath.ext
+    )
```

Comment:
Can we put the valid extensions in an array const at the top and reuse them via `.join(',')` in the other spots they show up?